--- sip/QtNetwork/qsslconfiguration.sip	2022-06-17 20:42:24.864351700 +0530
+++ sip/QtNetwork/qsslconfiguration.sip	2022-06-23 21:57:47.184264688 +0530
@@ -99,18 +99,6 @@ public:
     bool missingCertificateIsFatal() const;
     void setMissingCertificateIsFatal(bool cannotRecover);
     static const char *NextProtocolHttp1_1 /Encoding="None",NoSetter/;
-%If (Qt_6_3_0 -)
-    bool dtlsCookieVerificationEnabled() const;
-%End
-%If (Qt_6_3_0 -)
-    void setDtlsCookieVerificationEnabled(bool enable);
-%End
-%If (Qt_6_3_0 -)
-    static QSslConfiguration defaultDtlsConfiguration();
-%End
-%If (Qt_6_3_0 -)
-    static void setDefaultDtlsConfiguration(const QSslConfiguration &configuration);
-%End
 };
 
 %End
