diff --git a/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp b/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp
index 941d8df603d9..df28e9f21f2c 100644
--- a/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/QWebFrameAdapter.cpp
@@ -961,10 +961,9 @@ void QWebFrameAdapter::setViewportSize(const QSize& size)
     FrameView* view = frame->view();
     ASSERT(view);
     view->resize(size);
-    view->adjustViewSize();
-
     if (view->needsLayout())
         view->layout();
+    view->adjustViewSize();
 }
 
 
