--- src/corelib/io/qurl.cpp	2024-11-14 16:32:40.000000000 +0530
+++ src/corelib/io/qurl.cpp.new	2025-04-02 09:29:34.508658154 +0530
@@ -741,6 +741,32 @@ static const ushort * const pathInIsolat
 static const ushort * const queryInIsolation = userNameInIsolation + 6;
 static const ushort * const fragmentInIsolation = userNameInIsolation + 7;

+static const ushort localPathFromUser[] = {
+    // we force-decode some of the gen-delims, because
+    //    pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
+    // the gen-delim lines are leave() in qt_urlRecode, so we don't need to
+    // repeat them if we want to keep them decoded
+    // decode(':'), // allowed
+    // decode('@'), // allowed
+    encode(']'),
+    encode('['),
+    // decode('/'), // special and allowed
+    // decode('?'), // handled by path() and others
+    // decode('#'), // ditto
+
+    // the rest is like pathInIsolation above
+    decode('"'),
+    decode('<'),
+    decode('>'),
+    decode('^'),
+    decode('\\'),
+    decode('|'),
+    decode('{'),
+    decode('}'),
+
+    0
+};
+
 static const ushort userNameInUserInfo[] =  {
     encode(':'), // 0
     decode('@'), // 1
@@ -3341,7 +3367,11 @@ QUrl QUrl::fromLocalFile(const QString &
     }

     url.setScheme(scheme);
-    url.setPath(deslashified, DecodedMode);
+    // not directly using setPath here, as we do a few more transforms
+    parseDecodedComponent(deslashified);
+    if (!qt_urlRecode(url.d->path, deslashified, {}, localPathFromUser))
+        url.d->path = deslashified;
+
     return url;
 }
