diff --git a/src/gui/text/qtextimagehandler.cpp b/src/gui/text/qtextimagehandler.cpp
index 70e8961..5c56c30 100644
--- a/src/gui/text/qtextimagehandler.cpp
+++ b/src/gui/text/qtextimagehandler.cpp
@@ -23,12 +23,14 @@
 {
     // qt_findAtNxFile expects a file name that can be tested with QFile::exists.
     // so if the format.name() is a file:/ or qrc:/ URL, then we need to strip away the schema.
-    QString localFile = baseFileName;
-    if (localFile.startsWith("file:/"_L1))
-        localFile = localFile.sliced(6);
-    else if (localFile.startsWith("qrc:/"_L1))
-        localFile = localFile.sliced(3);
-
+    QString localFile;
+    const QUrl url(baseFileName);
+    if (url.isLocalFile())
+        localFile = url.toLocalFile();
+    else if (baseFileName.startsWith("qrc:/"_L1))
+        localFile = baseFileName.sliced(3);
+    else
+        localFile = baseFileName;
     extern QString qt_findAtNxFile(const QString &baseFileName, qreal targetDevicePixelRatio,
                                    qreal *sourceDevicePixelRatio);
     return qt_findAtNxFile(localFile, targetDevicePixelRatio, sourceDevicePixelRatio);
